p2p/host/relay/autorelay.go

```diff
@@ -127,6 +139,39 @@ func (ar *AutoRelay) background(ctx context.Context) {
 	}
 }
 
+func (ar *AutoRelay) refreshReservations() {
+	for p, info := range ar.relays {
+		if !info.resfreshAt.IsZero() && time.Now().After(info.resfreshAt) {
+			ar.mx.Lock()
+			// if we're no longer connected to the peer, there's nothing to do here.
+			if ar.host.Network().Connectedness(p) != network.Connected {
+				ar.mx.Unlock()
+				continue
+			}
+			ar.mx.Unlock()
+
+			// try reserving a slot again -> this is akin to a reservation refresh.
+			rsvp, err := circuitv2.Reserve(ar.ctx, ar.host, peer.AddrInfo{ID: p, Addrs: info.addrs})
+			if err != nil {
+				log.Debugf("error refreshing slot on relay: %s", err.Error())
+				continue
+			}
+
+			ar.mx.Lock()
+			info := &relayInfo{
+				addrs: rsvp.Addrs,
+			}
+			if rsvp.LimitDuration > MinTTLRefresh {
+				info.resfreshAt = time.Now().Add(rsvp.LimitDuration / 2)
+			}
```

Comment:
this is completely wrong -- the limit duration is the actual relay limit; you want to look at the `Expiration` which is when your reservation expires.